docs/book/v2/routing-middleware.md

```diff
@@ -0,0 +1,265 @@
+                'type' => Segment::class,
+                'options' => [
+                    'route' => '/album/:album_id',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => new PipeSpec(
+                            SelectLanguageMiddleware::class,
+                            AlbumFromRouteMiddelware::class,
+                            AlbumDetailHandler::class
+                        ),
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+## `MiddlewareInterface` or `RequestHandlerInterface` Service Strings
+
+Alternatively, if you only dispatch one class, you can pass a class-string or alias which resolves to a
+`MiddlewareInterface`, `RequestHandlerInterface` or `Laminas\Stratigility\MiddlewarePipe`.
+
+```php
+use Application\Handler\AlbumDetailHandler;
+use Application\Middleware\AlbumDetailMiddlewarePipe;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+
+return [
+    'router' => [
+        'routes' => [
+            'container-handler-class-literal' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/container-handler-class-literal',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => AlbumDetailHandler::class,
+                    ],
+                ],
+            ],
+            'container-handler' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/container-handler',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => 'container_id_for_album_detail_handler',
+                    ],
+                ],
+            ],
+            'container-pipe' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/container-pipe',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => AlbumDetailMiddlewarePipe::class,
+                    ],
+                ],
+            ],
+             'container-middleware' => [
+                 'type' => Literal::class,
+                 'options' => [
+                     'route' => '/container-middleware',
+                     'defaults' => [
+                         'controller' => PipeSpec::class,
+                         'middleware' => SomeMiddleware::class,
+                     ],
+                 ],
+             ],
+       ],
+    ],
+];
+```
+
+##  Anonymous Interface Implementations
+
+You can also pass an anonymous class which implements `MiddlewareInterface` or `RequestHandlerInterface` directly. This
+is primarily for development because it can cause issues with configuration caching.
```

Comment:
Missing word?

```suggestion
You can also pass an anonymous class which implements `MiddlewareInterface` or `RequestHandlerInterface` directly. This
is intended primarily for development because it can cause issues with configuration caching.
```